XenCert/StorageHandler.py

```diff
@@ -2981,6 +2981,210 @@ def MPConfigVerificationTests(self):
+            #    sout = util.pread2(cmd)
+            #    Print(sout)
+            #    displayOperationStatus(True)
+            #    checkPoints += 1
+            #except Exception, e:
+            #    Print("   - Failed to display shares on server: %s. Exception: %s" % (self.storage_conf['server'], str(e)))
+            #    raise e
+
+            # 2. Create directory and execute Filesystem IO tests
+            Print("CREATE CIFS SR AND PERFORM FILESYSTEM IO TESTS.")
+            Print(">> This test creates a CIFS SR and performs filesystem ")
+            Print(">> read write operations on the mounted directory.")
+            try:
+                # Create and plug SR
+                XenCertPrint( " Create CIFS SR.")
+                (retVal, sr_ref, _trash) = self.Create()
+                if not retVal:
+                    raise Exception("      SR creation failed.")
+                testSRCreated = True
+                testdir = "/var/run/sr-mount/%s/XenCertTestDir-%s" % (self.session.xenapi.SR.get_uuid(sr_ref), commands.getoutput('uuidgen'))
+
+                try:
+                    os.mkdir(testdir, 755)
+                except Exception,e:
+                    raise Exception("Exception creating directory: %s" % str(e))
+                testDirCreated = True
+                testfile = os.path.join(testdir, 'XenCertTestFile-%s' % commands.getoutput('uuidgen'))
+                cmd = ['dd', 'if=/dev/zero', 'of=%s' % testfile, 'bs=1M', 'count=1', 'oflag=direct']
+                (rc, stdout, stderr) = util.doexec(cmd, '')
+                testFileCreated = True
+                if rc != 0:
+                    raise Exception(stderr)
+                displayOperationStatus(True)
+                checkPoints += 1
+            except Exception, e:
+                Print("   - Failed to perform filesystem IO tests.")
+                raise e
+
+            # 3. Report Filesystem target space parameters for verification by user
+            Print("REPORT FILESYSTEM TARGET SPACE PARAMETERS FOR VERIFICATION BY THE USER")
+            try:
+                Print("  - %-20s: %s" % ('Total space', util.get_fs_size(testdir)))
+                Print("  - %-20s: %s" % ('Space utilization',util.get_fs_utilisation(testdir)))
+                displayOperationStatus(True)
+                checkPoints += 1
+            except Exception, e:
+                Print("   - Failed to report filesystem space utilization parameters. " )
+                raise e
+        except Exception, e:
+            Print("   - Functional testing failed with error: %s" % str(e))
+            retVal = False
+
+        # Now perform some cleanup here
+        try:
+            if testFileCreated:
+                os.remove(testfile)
+            if testDirCreated:
+                os.rmdir(testdir)
+            if testSRCreated:
+                StorageHandlerUtil.DestroySR(self.session, sr_ref)
+            checkPoints += 1
```

Comment:
Is it possible to perform this cleanup (and increment the checkPoint) only if the SR was created in the first place? Currently, this results in Functional tests being 33% successful even though they have completely failed.